test/priceUtils.js

```diff
@@ -26,5 +33,105 @@ contract("PriceOracle", function(accounts) {
+      const token2 = await ERC20.new("USDC", 6)
+
+      await prepareTokenRegistration(accounts[0], exchange)
+      await exchange.addToken(token1.address, { from: accounts[0] })
+      await prepareTokenRegistration(accounts[0], exchange)
+      await exchange.addToken(token2.address, { from: accounts[0] })
+      const USDCtokenId = (await exchange.numTokens.call()).toNumber() - 1
+
+      const orders = [
+        {
+          // normal order selling for more than 1
+          user: "0x4c7281e2bd549a0aea492b28ef60e3d81fed36e6",
+          sellTokenBalance: new BN("10eed9efc"),
+          buyToken: 1,
+          sellToken: USDCtokenId,
+          priceNumerator: new BN("101").mul(new BN(10).pow(new BN(18))),
+          priceDenominator: new BN("100").mul(new BN(10).pow(new BN(6))),
+        },
+        {
+          // normal order selling for more than 1
+          user: "0x4c7281e2bd549a0aea492b28ef60e3d81fed36e6",
+          sellTokenBalance: new BN("0"),
+          buyToken: USDCtokenId,
+          sellToken: 1,
+          priceNumerator: new BN("101").mul(new BN(10).pow(new BN(6))),
+          priceDenominator: new BN("100").mul(new BN(10).pow(new BN(18))),
+        },
+      ]
+      assert.equal(await checkNoProfitableOffer(orders[0], exchange, 1.0), true)
+      assert.equal(await checkNoProfitableOffer(orders[1], exchange, 1.0), true)
+    })
+    it("checks that bracket traders does not sell unprofitable for tokens with the different decimals", async () => {
+      const ERC20 = artifacts.require("DetailedMintableToken")
+      const token1 = await ERC20.new("WETH", 18)
+      const token2 = await ERC20.new("USDC", 6)
+
+      await prepareTokenRegistration(accounts[0], exchange)
+      await exchange.addToken(token1.address, { from: accounts[0] })
+      await prepareTokenRegistration(accounts[0], exchange)
+      await exchange.addToken(token2.address, { from: accounts[0] })
+      const USDCtokenId = (await exchange.numTokens.call()).toNumber() - 1
+      const orders = [
+        {
+          // order is profitable for others
+          user: "0x4c7281e2bd549a0aea492b28ef60e3d81fed36e6",
+          sellTokenBalance: new BN("10eed9efc"),
+          buyToken: 1,
+          sellToken: USDCtokenId,
+          priceNumerator: new BN("99").mul(new BN(10).pow(new BN(18))),
+          priceDenominator: new BN("100").mul(new BN(10).pow(new BN(6))),
+        },
+        {
+          // order is profitable for others, but balance is 0
+          user: "0x4c7281e2bd549a0aea492b28ef60e3d81fed36e6",
+          sellTokenBalance: new BN("0"),
+          buyToken: 1,
+          sellToken: USDCtokenId,
+          priceNumerator: new BN("101").mul(new BN(10).pow(new BN(18))),
+          priceDenominator: new BN("100").mul(new BN(10).pow(new BN(6))),
+        },
+      ]
+      assert.equal(await checkNoProfitableOffer(orders[0], exchange, 1), false)
+      assert.equal(await checkNoProfitableOffer(orders[1], exchange, 1), true)
+    })
```

Comment:
Since the code is almost the same as in the test above (it took me a while to spot the difference!) maybe it would be clearer if we had the same function called on each test instead of repeating the code?

Reply:
If we are comparing all 3 added tests, we have: token creations( individual), token additions to the exchange(every time the same code) and order definition(individual) and final check (individual).

Since the token additions are the only code that really is the same, I could factor this into a function. But since it is only 4 lines, I would say that it is fine.
I think the readability improves from this easy 'log book' style.